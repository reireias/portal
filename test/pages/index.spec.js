import { mount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import tartget from '@/pages/index.vue'

const localVue = createLocalVue()

Vue.use(Vuetify)
localVue.use(Vuex)
localVue.use(Vuetify)

describe('index', () => {
  let actions
  let getters
  let store
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
    actions = {
      bindMessages: jest.fn(),
    }
    getters = {
      messages: jest.fn(),
    }
    store = new Vuex.Store({
      state: {},
      getters,
      actions,
    })
  })

  test('is a Vue instance', () => {
    const wrapper = mount(tartget, {
      localVue,
      store,
      vuetify,
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(actions.bindMessages).toHaveBeenCalled()
    expect(getters.messages).toHaveBeenCalled()
  })
})
