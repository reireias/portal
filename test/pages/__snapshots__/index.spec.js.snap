// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index is a Vue instance 1`] = `
"<div class=\\"container\\">
  <div class=\\"row justify-center\\">
    <div class=\\"col-md-8 col-12\\">
      <div class=\\"container\\"><span></span></div>
    </div>
  </div>
</div>"
`;
